tests/publiccloud/ssh_interactive_init.pm

```diff
@@ -39,49 +34,44 @@ sub run {
     $args->{my_instance} = $instance;
 
     # configure ssh client, fetch the instance ssh public key, do not use default $instance->ssh_opts
-    assert_script_run('curl ' . data_url('publiccloud/ssh_config') . ' -o ~/.ssh/config');
-    assert_script_run "ssh-keyscan " . $instance->public_ip . " >> ~/.ssh/known_hosts";
+    my $ssh_config = data_url('publiccloud/ssh_config');
+    assert_script_run("curl $ssh_config -o ~/.ssh/config");
+    assert_script_run("ssh-keyscan $instance->{public_ip} >> ~/.ssh/known_hosts");
     $instance->ssh_opts("");
 
     # Create the ssh alias
-    assert_script_run('echo -en "Host sut\n  Hostname ' . $instance->public_ip . '\n" >> ~/.ssh/config');
-
-    # Copy the SSH settings also for normal user
-    assert_script_run("mkdir /home/" . $testapi::username . "/.ssh");
-    assert_script_run("chmod -R 700 /home/" . $testapi::username . "/.ssh/");
-    assert_script_run("cp ~/.ssh/* /home/" . $testapi::username . "/.ssh/");
-    assert_script_run("chown -R " . $testapi::username . " /home/" . $testapi::username . "/.ssh/");
-
-    # configure ssh server, allow root and password login
-    $instance->run_ssh_command(cmd => 'hostname');
-    $instance->run_ssh_command(cmd => 'sudo sed -i "s/PasswordAuthentication/#PasswordAuthentication/" /etc/ssh/sshd_config; echo "PasswordAuthentication no" | sudo tee -a /etc/ssh/sshd_config');
-    $instance->run_ssh_command(cmd => 'sudo sed -i "s/ChallengeResponseAuthentication/#ChallengeResponseAuthentication/" /etc/ssh/sshd_config; echo "ChallengeResponseAuthentication no" | sudo tee -a /etc/ssh/sshd_config');
+    assert_script_run(qq(echo -e "Host sut\\n  Hostname $instance->{public_ip}" >> ~/.ssh/config));
+
+    # Copy SSH settings also for normal user
+    assert_script_run("install -o $testapi::username -g users -m 0700 -dD /home/$testapi::username/.ssh");
+    assert_script_run("install -o $testapi::username -g users -m 0600 ~/.ssh/* /home/$testapi::username/.ssh/");
+
+    # DEBUG !!!
```

Comment:
maybe we need to define something like `if(get('PUBLIC_CLOUD_DEBUG'))` ? so we can switch on and off this at runtime ?

Reply:
I planned to remove this but I can probably do so but I can do so if you find it useful in OSD.

Reply:
than just remove it